controllers/ExportController.php

```diff
@@ -0,0 +1,146 @@
+use humhub\modules\post\models\Post;
+use humhub\modules\content\models\Content;
+use humhub\modules\comment\models\Comment;
+use yii\web\Response;
+
+/**
+ * ExportController handles exporting user data as JSON.
+ */
+class ExportController extends BaseAccountController
+{
+    /**
+     * Renders the view displaying the exported user data as JSON.
+     *
+     * @return string The rendered view.
+     */
+    public function actionIndex()
+    {
+        return $this->renderAjax('index');
+    }
+
+    /**
+     * Downloads the exported user data as a JSON file.
+     *
+     * @return Response The file response.
+     */
+    public function actionDownload()
+    {
+        // Check if the REST module is enabled
+        if (!Yii::$app->hasModule('rest')) {
+            throw new \yii\base\Exception('REST module is not enabled.');
+        }
+
+        $userData = $this->getUserData();
+        $postData = $this->getPostData();
+        $contentData = $this->getContentData();
+        $commentData = $this->getCommentData();
+
+        // Combine user, post, content, and comment data
+        $data = [
+            'user' => $userData,
+            'post' => $postData,
+            'content' => $contentData,
+            'comment' => $commentData,
+        ];
+
+        // Convert data to JSON
+        $jsonData = json_encode($data, JSON_PRETTY_PRINT);
+
+        // Set headers for JSON file download
+        Yii::$app->response->format = Response::FORMAT_RAW;
+        Yii::$app->response->headers->add('Content-Type', 'application/json');
+        Yii::$app->response->headers->add('Content-Disposition', 'attachment; filename="userdata.json"');
+
+        // Output JSON data to response body
+        Yii::$app->response->content = $jsonData;
+
+        // Send the response
+        return Yii::$app->response;
+    }
+
+    /**
+     * Retrieves user data.
+     *
+     * @return array The user data.
+     */
+    private function getUserData()
+    {
+        $currentUser = Yii::$app->user->getIdentity();
+        if (Yii::$app->hasModule('rest')) {
+            return UserDefinitions::getUser($currentUser);
+        } else {
+            return $currentUser->attributes;
+        }
+    }
+
+    /**
+     * Retrieves current user's posts and returns as JSON.
+     *
+     * @return array The JSON data.
+     */
+    private function getPostData()
+    {
+        $currentUser = Yii::$app->user->getIdentity();
+        if (Yii::$app->hasModule('rest')) {
+            $userPosts = Post::find()
+                ->where(['created_by' => $currentUser->id])
+                ->all();
+            return array_map(function($post) {
+                return PostDefinitions::getPost($post);
+            }, $userPosts);
+        } else {
+            return [];
+        }
+    }
+
+    /**
+     * Retrieves current user's content and returns as JSON.
+     *
+     * @return array The JSON data.
+     */
+    private function getContentData()
```

Comment:
It's safe to say that I can remove this and use a different definition?

Reply:
Which definition?

Reply:
More than likely Files and Likes as these two fit the requirements.

Reply:
We should not add the Content MetaData in the Data Export module, instead we should add the actual content (e.g. the posts). These automatically include the content metadata.

Reply:
To to break thinks down the following APIs should be used;

We could probably get away with not using Like, Topics or File APIs as they're also included in the Content API which the metadata is gotten through the Post API, but if it's wanted I can add this as we can't be sure that all Likes or Files are gotten for other types.

- [User API](https://github.com/humhub/rest/blob/master/definitions/UserDefinitions.php)
- [Post API](https://github.com/humhub/rest/blob/master/definitions/PostDefinitions.php)
- [Like API](https://github.com/humhub/rest/blob/master/definitions/LikeDefinitions.php)
- [File API](https://github.com/humhub/rest/blob/master/definitions/FileDefinitions.php)
- [Comment API](https://github.com/humhub/rest/blob/master/definitions/CommentDefinitions.php)

If there are any others that are required I can have them added as well.